src/Microsoft.TestPlatform.CommunicationUtilities/SocketClient.cs

```diff
@@ -56,8 +62,28 @@ public string Start(string endPoint)
                 EqtTrace.Verbose("Waiting for connecting to server");
             }
 
-            // Don't start if the endPoint port is zero
-            this.tcpClient.ConnectAsync(ipEndPoint.Address, ipEndPoint.Port).ContinueWith(this.OnServerConnected);
+            Task connectionTask = null;
+
+            Stopwatch watch = new Stopwatch();
+            watch.Start();
+
+            // It is possible that first ConnectAsync call could not connect to server, so do a repoll to connect to server for 50secs
+            do
+            {
+                try
+                {
+                    connectionTask = this.tcpClient.ConnectAsync(ipEndPoint.Address, ipEndPoint.Port);
+                    connectionTask.Wait();
```

Comment:
Should we sleep for few ms before retry ?

Reply:
We don't need to, ConnectAsync is an async call, & it does not time out immediately.